File: src/exp_client.py
#!/bin/env python
#
# Example client
#
# Usage: exp_client.py ENDPOINT MESSAGE ...
#
# % python exp_client.py tcp://127.0.0.1:5557 message
# % python exp_client.py tcp://127.0.0.1:5557 message1 message2 message3

import sys
import os
import zmq

argv = sys.argv
argc = len(argv)

if argc <= 2:
    print "Usage: python %s ENDPOINT MESSAGE ..." % os.path.basename(__file__)
    exit(1)

endpoint = argv[1]

context = zmq.Context()
socket = context.socket(zmq.PUSH)
socket.connect(endpoint)

last = argc - 1
i = 2

while i != last:
    socket.send(str(argv[i]), zmq.SNDMORE)
    i = i + 1

socket.send(str(argv[last]))

File: src/exp_client.rb
#!/bin/env ruby
#
# Example client
#
# Usage: exp_client.rb ENDPOINT MESSAGE ...
#
# % ruby exp_client.rb tcp://127.0.0.1:5557 message
# % ruby exp_client.rb tcp://127.0.0.1:5557 message1 message2 message3

require 'zmq'

if ARGV.length <= 1 then
  print "Usage: ruby ", File.basename(__FILE__), " ENDPOINT MESSAGE ...\n"
  exit
end

endpoint = ARGV[0]

context = ZMQ::Context.new(1)
socket = context.socket(ZMQ::PUSH)
socket.connect(endpoint)

last = ARGV.length - 1
i = 1

while i != last
  socket.send(ARGV[i].to_s, ZMQ::SNDMORE)
  i = i + 1
end

socket.send(ARGV[last].to_s)

socket.close()
context.close()
